- 금별맥주 : 원본(raw, 기준 데이터)
- 금별맥주2 : 원본 복사본 + 중복 관련 실험/정제 진행 중
- 금별맥주3 : 중복 place_id만 추출, 지점명 = NaN or 중앙점 삭제
- 금별맥주4 : 중복 place_id에 대해 정제완료
- 금별맥주5 : 금별맥주2 + 금별맥주4 = 정제완료

In [3]:
import pandas as pd
import numpy as np

pd.set_option('display.max_columns', None) # 전역 옵션 설정

df= pd.read_csv( '../소상공인시장진흥공단_상가정보_추가수집데이터_2.csv')

# brand가 '금별맥주' 인것만 출력
금별맥주 = df[df['brand'] == '금별맥주']
금별맥주

# place_id 정수로 변경하기 -> 완료
# 층정보, 지번부번지 결측값 건들지말기


상가업소번호 brand         상호명   지점명 상권업종대분류코드 상권업종대분류명  \
9488   MA0101202209A0100150  금별맥주        금별맥주   NaN        I2       음식   
9489   MA010120220804156426  금별맥주  금별맥주동양미래대점   NaN        I2       음식   
9494   MA010120220808688996  금별맥주   금별맥주양천향교점   NaN        I2       음식   
9498   MA010120220806941672  금별맥주        금별맥주   NaN        I2       음식   
9504   MA010120220809845042  금별맥주        금별맥주   NaN        I2       음식   
...                     ...   ...         ...   ...       ...      ...   
10304  MA010120220812082878  금별맥주   금별맥주별내파라곤  스퀘어점        I2       음식   
10323  MA010120220808415184  금별맥주        금별맥주   NaN        I2       음식   
10327  MA0101202209A0010612  금별맥주     금별맥주모란점   NaN        I2       음식   
10334  MA010120220803359144  금별맥주        금별맥주   NaN        I2       음식   
10336  MA010120220813083656  금별맥주        금별맥주   NaN        I2       음식   

      상권업종중분류코드 상권업종중분류명 상권업종소분류코드 상권업종소분류명 표준산업분류코드  표준산업분류명  시도코드    시도명  \
9488       I211       주점    I21103   생맥주 전문   I56213  생맥주 전문점    11  서울특별시   
9489       I211       주점    I21103   생맥주 전문   I56219   기타 주점업    11  서울특별시   
9494       I211       주점    I21103   생맥주 전문   I56213  생맥주 전문점    11  서울특별시   
9498       I211       주점    I21103   생맥주 전문   I56213  생맥주 전문점    11  서울특별시   
9504       I211       주점    I21103   생맥주 전문   I56219   기타 주점업    11  서울특별시   
...         ...      ...       ...      ...      ...      ...   ...    ...   
10304      I211       주점    I21103   생맥주 전문   I56219   기타 주점업    41    경기도   
10323      I211       주점    I21103   생맥주 전문   I56219   기타 주점업    41    경기도   
10327      I211       주점    I21103   생맥주 전문   I56219   기타 주점업    41    경기도   
10334      I211       주점    I21104    요리 주점   I56219   기타 주점업    41    경기도   
10336      I211       주점    I21103   생맥주 전문   I56213  생맥주 전문점    41    경기도   

       시군구코드     시군구명     행정동코드  행정동명       법정동코드 법정동명                 지번코드  \
9488   11470      양천구  11470650  신정4동  1147010100  신정동  1147010100108870008   
9489   11530      구로구  11530720  고척1동  1153010600  고척동  1153010600100620134   
9494   11500      강서구  11500603  가양1동  1150010500  마곡동  1150010500107760000   
9498   11710      송파구  11710642  문정2동  1171010800  문정동  1171010800106430001   
9504   11650      서초구  11650520  서초2동  1165010800  서초동  1165010800113270035   
...      ...      ...       ...   ...         ...  ...                  ...   
10304  41360     남양주시  41360570   별내동  4136011100  별내동  4136011100109750001   
10323  41370      오산시  41370523  대원1동  4137010300   원동  4137010300108120006   
10327  41133  성남시 중원구  41133510   성남동  4113310100  성남동  4113310100134430000   
10334  41210      광명시  41210655   일직동  4121010600  일직동  4121010600105120003   
10336  41196  부천시 오정구  41196530  원종2동  4119610400  원종동  4119610400102270013   

        지번본번지  지번부번지                    지번주소         도로명코드  \
9488    887.0    8.0     서울특별시 양천구 신정동 887-8  114703114011   
9489     62.0  134.0    서울특별시 구로구 고척동 62-134  115304148103   
9494    776.0    NaN       서울특별시 강서구 마곡동 776  115004145568   
9498    643.0    1.0     서울특별시 송파구 문정동 643-1  117103350849   
9504   1327.0   35.0   서울특별시 서초구 서초동 1327-35  116504163436   
...       ...    ...                     ...           ...   
10304   975.0    1.0      경기도 남양주시 별내동 975-1  413603197037   
10323   812.0    6.0        경기도 오산시 원동 812-6  413703198034   
10327  3443.0    NaN    경기도 성남시 중원구 성남동 3443  411334337420   
10334   512.0    3.0       경기도 광명시 일직동 512-3  412103187066   
10336   227.0   13.0  경기도 부천시 오정구 원종동 227-13  411964358356   

                       도로명                   도로명주소  층정보          경도  \
9488       서울특별시 양천구 신정중앙로      서울특별시 양천구 신정중앙로 95    1  126.862547   
9489      서울특별시 구로구 경인로47길      서울특별시 구로구 경인로47길 8    2  126.867221   
9494     서울특별시 강서구 마곡동로10길    서울특별시 강서구 마곡동로10길 23  NaN  126.839288   
9498         서울특별시 송파구 법원로       서울특별시 송파구 법원로 114  NaN  127.120090   
9504     서울특별시 서초구 서초대로78길    서울특별시 서초구 서초대로78길 38    2  127.028225   
...                    ...                     ...  ...         ...   
1030

In [4]:
금별맥주.info()

<class 'pandas.core.frame.DataFrame'>
Index: 118 entries, 9488 to 10336
Data columns (total 34 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   상가업소번호     118 non-null    object 
 1   brand      118 non-null    object 
 2   상호명        118 non-null    object 
 3   지점명        62 non-null     object 
 4   상권업종대분류코드  118 non-null    object 
 5   상권업종대분류명   118 non-null    object 
 6   상권업종중분류코드  118 non-null    object 
 7   상권업종중분류명   118 non-null    object 
 8   상권업종소분류코드  118 non-null    object 
 9   상권업종소분류명   118 non-null    object 
 10  표준산업분류코드   118 non-null    object 
 11  표준산업분류명    118 non-null    object 
 12  시도코드       118 non-null    int64  
 13  시도명        118 non-null    object 
 14  시군구코드      118 non-null    int64  
 15  시군구명       118 non-null    object 
 16  행정동코드      118 non-null    int64  
 17  행정동명       118 non-null    object 
 18  법정동코드      118 non-null    int64  
 19  법정동명       118 non-null    object 
 20  지번코드      

In [5]:
# 타입변환 .astype(타입)
# .astype(타입) => 결측값 있으면 오류남

# 소숫점 제거를 위해 int로 변환
금별맥주['place_id'] = 금별맥주['place_id'].astype('int')
금별맥주['place_id']

C:\Users\user\AppData\Local\Temp\ipykernel_10480\3420801769.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  금별맥주['place_id'] = 금별맥주['place_id'].astype('int')


9488     1265782586
9489     1113384749
9494     1320051621
9498     1588254549
9504     1494938619
            ...    
10304       2360111
10323    1904029520
10327    1976292314
10334    1418180719
10336    1284647505
Name: place_id, Length: 118, dtype: int64

In [6]:
'''<class 'pandas.core.frame.DataFrame'>
Index: 118 entries, 9488 to 10336
Data columns (total 34 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   상가업소번호     118 non-null    object 
 1   brand      118 non-null    object 
 2   상호명        118 non-null    object 
 3   지점명        62 non-null     object 
 4   상권업종대분류코드  118 non-null    object 
 5   상권업종대분류명   118 non-null    object 
 6   상권업종중분류코드  118 non-null    object 
 7   상권업종중분류명   118 non-null    object 
 8   상권업종소분류코드  118 non-null    object 
 9   상권업종소분류명   118 non-null    object 
 10  표준산업분류코드   118 non-null    object 
 11  표준산업분류명    118 non-null    object 
 12  시도코드       118 non-null    int64  
 13  시도명        118 non-null    object 
 14  시군구코드      118 non-null    int64  
 15  시군구명       118 non-null    object 
 16  행정동코드      118 non-null    int64  
 17  행정동명       118 non-null    object 
 18  법정동코드      118 non-null    int64  
 19  법정동명       118 non-null    object 
 20  지번코드       118 non-null    int64  
 21  지번본번지      118 non-null    float64
 22  지번부번지      97 non-null     float64
 23  지번주소       118 non-null    object 
 24  도로명코드      118 non-null    int64  
 25  도로명        118 non-null    object 
 26  도로명주소      118 non-null    object 
 27  층정보        81 non-null     object 
 28  경도         118 non-null    float64
 29  위도         118 non-null    float64
 30  place_id   118 non-null    int64           -> int로 변경된거 확인완료
 31  월간 검색량     118 non-null    float64
 32  블로그누적발행    118 non-null    float64
 33  encoded    118 non-null    object 
dtypes: float64(6), int64(7), object(21)
memory usage: 32.3+ KB'''

"<class 'pandas.core.frame.DataFrame'>\nIndex: 118 entries, 9488 to 10336\nData columns (total 34 columns):\n #   Column     Non-Null Count  Dtype  \n---  ------     --------------  -----  \n 0   상가업소번호     118 non-null    object \n 1   brand      118 non-null    object \n 2   상호명        118 non-null    object \n 3   지점명        62 non-null     object \n 4   상권업종대분류코드  118 non-null    object \n 5   상권업종대분류명   118 non-null    object \n 6   상권업종중분류코드  118 non-null    object \n 7   상권업종중분류명   118 non-null    object \n 8   상권업종소분류코드  118 non-null    object \n 9   상권업종소분류명   118 non-null    object \n 10  표준산업분류코드   118 non-null    object \n 11  표준산업분류명    118 non-null    object \n 12  시도코드       118 non-null    int64  \n 13  시도명        118 non-null    object \n 14  시군구코드      118 non-null    int64  \n 15  시군구명       118 non-null    object \n 16  행정동코드      118 non-null    int64  \n 17  행정동명       118 non-null    object \n 18  법정동코드      118 non-null    int64  \n 19  법정동명       118 non-null  

In [7]:
# place_id 중복 여부 확인
print(금별맥주['place_id'].nunique(), len(금별맥주))

# place_id 중복된 갯수 확인
print("place_id 중복:", 금별맥주['place_id'].duplicated().sum())

107 118
place_id 중복: 11


In [8]:
pd.set_option('display.max_rows', None)

금별맥주.loc[
    금별맥주['place_id'].duplicated(keep=False)
].sort_values('place_id')[['place_id', '상호명', '지점명']]

place_id          상호명       지점명
9699   1035417750         금별맥주  구로디지털단지점
9635   1035417750         금별맥주      신도림점
9866   1175386565         금별맥주       NaN
10178  1175386565         금별맥주       미금점
9759   1195530733         금별맥주       NaN
9634   1195530733         금별맥주      사당역점
9717   1249032365         금별맥주       NaN
9740   1249032365     금별맥주마곡역점       NaN
9576   1265782586  금별맥주신정네거리역점       NaN
9488   1265782586         금별맥주       NaN
10334  1418180719         금별맥주       NaN
10059  1418180719         금별맥주      광명역점
10119  1473371787    금별맥주한대앞역점       NaN
10071  1473371787       금별맥주안산       중앙점
9883   1473371787         금별맥주       NaN
9504   1494938619         금별맥주       NaN
9648   1494938619         금별맥주       NaN
10292  1769471357   금별맥주수원고색역점       NaN
10226  1769471357         금별맥주      오목천점
9930   1987203694         금별맥주       야당점
10129  1987203694    금별맥주파주초롱꽃       마을점

In [9]:
# 1035417750 -> 구로디지털단지점
# 1175386565 -> 분당 미금점
# 1195530733 -> 사당점	 
# 1249032365 -> 마곡역점
# 1265782586 -> 신정네거리역점
# 1418180719 -> 광명역점
# 1473371787 -> 안산한대앞역점
# 1494938619 -> 삼성타운점
# 1769471357 -> 수원고색역점	
# 1987203694 -> 파주초롱꽃마을점

#==================================
# PXMot
# 신도림점 -> 1189283598
# 안산중앙점 -> 폐점
# 오목천점 -> 1074672431
# 야당점 -> 1021328867

### ***중복처리 순서***
1. **중복 place_id 처리**
    - place_id_fix 사전을 기준으로 중복 place_id 중 올바른 지점만 남기고 나머지는 NaN 처리
    - 잘못된 중복을 제거하기 전에 모든 행을 살려두는 것이 중요

2. **신규 지점 place_id 지정**

    2-1.**처음 데이터에는 중복 place_id가 있음**
    - 예: 1035417750이라는 place_id가 구로디지털단지점과 신도림점에 동시에 존재
    - 즉, 중복 때문에 원래 의도한 지점과 일치하지 않음

    2-2. **place_id를 확인/검색해서 정확한 지점 정보 확보**
    - 예를 들어 신도림점은 실제 place_id가 1189283598임 → 기존 중복 place_id와 다름
    
    2-3. **정확한 매칭을 위해 기존 place_id를 교체**
    - 기존 place_id가 잘못 들어간 행 → 새로 확인한 정확한 place_id로 변경
    - 이렇게 해야 크롤링이나 분석할 때 실제 매장과 데이터가 일치

    2-4. **데이터 일관성 확보**
    - 모든 지점이 고유 place_id를 갖게 되면
    - 중복/오류로 인한 혼동 제거
    - NaN 또는 잘못된 데이터 없이 분석 가능

3. **place_id dtype 통일**
    - NaN이 존재할 수 있으므로 **nullable integer (Int64)**로 변환

4. **폐점 및 잘못된 중복 행 제거**
    - place_id가 NaN인 행 → 중복 처리 실패 또는 존재하지 않는 place_id
    - 폐점 지점 제거 (예: 안산중앙점)

5. **최종 확인**
    - 모든 place_id가 **정수(Int64)**이고
    - 중복 없이 올바른 지점만 남았는지 확인

In [10]:
# 기존 금별맥주 복사
금별맥주2 = 금별맥주.copy()  # 원본 데이터 보존

In [11]:
# 현재 place_id 중복 데이터프레임 확인
pd.set_option('display.max_rows', None)

금별맥주2.loc[
    금별맥주2['place_id'].duplicated(keep=False)
].sort_values('place_id')[['place_id', '상호명', '지점명']]

place_id          상호명       지점명
9699   1035417750         금별맥주  구로디지털단지점
9635   1035417750         금별맥주      신도림점
9866   1175386565         금별맥주       NaN
10178  1175386565         금별맥주       미금점
9759   1195530733         금별맥주       NaN
9634   1195530733         금별맥주      사당역점
9717   1249032365         금별맥주       NaN
9740   1249032365     금별맥주마곡역점       NaN
9576   1265782586  금별맥주신정네거리역점       NaN
9488   1265782586         금별맥주       NaN
10334  1418180719         금별맥주       NaN
10059  1418180719         금별맥주      광명역점
10119  1473371787    금별맥주한대앞역점       NaN
10071  1473371787       금별맥주안산       중앙점
9883   1473371787         금별맥주       NaN
9504   1494938619         금별맥주       NaN
9648   1494938619         금별맥주       NaN
10292  1769471357   금별맥주수원고색역점       NaN
10226  1769471357         금별맥주      오목천점
9930   1987203694         금별맥주       야당점
10129  1987203694    금별맥주파주초롱꽃       마을점

In [12]:
# 1. 중복 place_id 수정 매핑
place_id_fix = {
    1035417750: "구로디지털단지점",
    1175386565: "분당미금점",
    1195530733: "사당점",
    1249032365: "마곡역점",
    1265782586: "신정네거리역점",
    1418180719: "광명역점",
    1473371787: "안산한대앞역점",
    1494938619: "삼성타운점",
    1769471357: "수원고색역점",
    1987203694: "파주초롱꽃마을점"
}

# place_id_fix를 데이터프레임으로 변환해서 확인하기
fix_df = (
    pd.DataFrame(
        place_id_fix.items(),
        columns=['place_id', '지점명_fix']
    )
)

In [13]:
# 원본 데이터와 join해서 비교하기
# 현재 들어있는 값과 의도한 수정값 한눈에 확인 가능
check_df = (
    금별맥주2
    .merge(fix_df, on='place_id', how='inner')
    [['place_id', '지점명', '지점명_fix', '지번주소']]
)

display(check_df)

place_id       지점명   지점명_fix                    지번주소
0   1265782586       NaN   신정네거리역점     서울특별시 양천구 신정동 887-8
1   1494938619       NaN     삼성타운점   서울특별시 서초구 서초동 1327-35
2   1265782586       NaN   신정네거리역점      서울특별시 양천구 신정동 1326
3   1195530733      사당역점       사당점    서울특별시 서초구 방배동 444-16
4   1035417750      신도림점  구로디지털단지점        서울특별시 구로구 구로동 30
5   1494938619       NaN     삼성타운점    서울특별시 서초구 서초동 1574-4
6   1035417750  구로디지털단지점  구로디지털단지점   서울특별시 구로구 구로동 1124-38
7   1249032365       NaN      마곡역점     서울특별시 강서구 마곡동 797-9
8   1249032365       NaN      마곡역점     서울특별시 강서구 마곡동 799-2
9   1195530733       NaN       사당점       서울특별시 동작구 사당동 138
10  1175386565       NaN     분당미금점   경기도 성남시 분당구 정자동 159-4
11  1473371787       NaN   안산한대앞역점   경기도 안산시 상록구 본오동 879-6
12  1987203694       야당점  파주초롱꽃마을점        경기도 파주시 야당동 1085
13  1418180719      광명역점      광명역점       경기도 광명시 일직동 512-3
14  1473371787       중앙점   안산한대앞역점  경기도 안산시 단원구 고잔동 539-18
15  1473371787       NaN   안산한대앞역점    경기도 안산시 상록구 이동 715-2
16  1987203694       마을점  파주초롱꽃마을점      경기도 파주시 동패동 2080-1
17  1175386565       미금점     분당미금점    경기도 성남시 분당구 구미동 41-2
18  1769471357      오목천점    수원고색역점  경기도 수원시 권선구 오목천동 551-2
19  1769471357       NaN    수원고색역점   경기도 수원시 권선구 고색동 405-1
20  1418180719       NaN      광명역점       경기도 광명시 일직동 512-3

In [14]:
# 2. 신규 지점 place_id 지정
금별맥주2.loc[금별맥주2['지점명'] == '신도림점', 'place_id'] = 1189283598
금별맥주2.loc[금별맥주2['지점명'] == '오목천점', 'place_id'] = 1074672431
금별맥주2.loc[금별맥주2['지점명'] == '야당점', 'place_id'] = 1021328867

In [15]:
# 3. place_id dtype 통일 (nullable integer)
금별맥주2['place_id'] = 금별맥주2['place_id'].astype('Int64')

In [16]:
# 신규 place_id, 지번주소, 도로명 주소 일치한지 확인
check_ids = [1189283598, 1074672431, 1021328867]
display(금별맥주2[금별맥주2['place_id'].isin(check_ids)][['지점명','place_id', '지번주소', '도로명주소']])

지점명    place_id                    지번주소                도로명주소
9635   신도림점  1189283598        서울특별시 구로구 구로동 30  서울특별시 구로구 새말로16길 15
9930    야당점  1021328867        경기도 파주시 야당동 1085      경기도 파주시 소리천로 29
10226  오목천점  1074672431  경기도 수원시 권선구 오목천동 551-2  경기도 수원시 권선구 오목천로 61

In [17]:
# 폐점 지점 제거 (예: 안산중앙점)
# 금별맥주2 = 금별맥주2[금별맥주2['지점명'] != '중앙점'].reset_index(drop=True)

# 상호명과 지점명을 알 수 없을 때 -> 좌표로 지점 찾기

In [18]:
금별맥주2.loc[
    금별맥주2['place_id'].duplicated(keep=False)
].sort_values('place_id')[['place_id', '상호명', '지점명','지번주소', '도로명주소','위도','경도']]


place_id          상호명   지점명                    지번주소  \
10178  1175386565         금별맥주   미금점    경기도 성남시 분당구 구미동 41-2   
9866   1175386565         금별맥주   NaN   경기도 성남시 분당구 정자동 159-4   
9634   1195530733         금별맥주  사당역점    서울특별시 서초구 방배동 444-16   
9759   1195530733         금별맥주   NaN       서울특별시 동작구 사당동 138   
9717   1249032365         금별맥주   NaN     서울특별시 강서구 마곡동 797-9   
9740   1249032365     금별맥주마곡역점   NaN     서울특별시 강서구 마곡동 799-2   
9488   1265782586         금별맥주   NaN     서울특별시 양천구 신정동 887-8   
9576   1265782586  금별맥주신정네거리역점   NaN      서울특별시 양천구 신정동 1326   
10334  1418180719         금별맥주   NaN       경기도 광명시 일직동 512-3   
10059  1418180719         금별맥주  광명역점       경기도 광명시 일직동 512-3   
10071  1473371787       금별맥주안산   중앙점  경기도 안산시 단원구 고잔동 539-18   
9883   1473371787         금별맥주   NaN   경기도 안산시 상록구 본오동 879-6   
10119  1473371787    금별맥주한대앞역점   NaN    경기도 안산시 상록구 이동 715-2   
9504   1494938619         금별맥주   NaN   서울특별시 서초구 서초동 1327-35   
9648   1494938619         금별맥주   NaN    서울특별시 서초구 서초동 1574-4   

                         도로명주소         위도          경도  
10178  경기도 성남시 분당구 미금일로86번길 20  37.347801  127.110480  
9866      경기도 성남시 분당구 정자일로 162  37.364627  127.106305  
9634         서울특별시 서초구 동작대로 18  37.478103  126.982092  
9759     서울특별시 동작구 동작대로27나길 21  37.486608  126.980676  
9717       서울특별시 강서구 마곡중앙6로 70  37.559854  126.834848  
9740        서울특별시 강서구 공항대로 190  37.559034  126.828636  
9488        서울특별시 양천구 신정중앙로 95  37.526607  126.862547  
9576       서울특별시 양천구 중앙로43길 14  37.520421  126.851337  
10334           경기도 광명시 양지로 21  37.417687  126.882211  
10059           경기도 광명시 양지로 19  37.419012  126.882615  
10071      경기도 안산시 단원구 고잔1길 45  37.318396  126.838525  
9883       경기도 안산시 상록구 샘골로 165  37.298631  126.862918  
10119     경기도 안산시 상록구 광덕1로 379  37.308655  126.851248  
9504      서울특별시 서초구 서초대로78길 38  37.494526  127.028225  
9648      서울특별시 서초구 서초대로50길 19  37.492403  127.013162

In [19]:
# 경기도 성남시 분당구 정자일로 162	    37.364627	127.106305 -> x
# 서울특별시 동작구 동작대로27나길 21	37.486608	126.980676 -> 중복 -> 삭제하기
# 서울특별시 강서구 마곡동 797-9	    37.559854	126.834848 -> 발산역점
# 서울특별시 양천구 신정중앙로 95	    37.526607	126.862547 -> 목동로데오점
# 경기도 광명시 양지로 21	            37.417687	126.882211 -> 중복 -> 삭제하기
# 경기도 안산시 단원구 고잔1길 45	    37.318396	126.838525 -> x
# 경기도 안산시 상록구 샘골로 165	    37.298631	126.862918 -> x
# 서울특별시 서초구 서초대로78길38      37.494526	127.028225 -> 삼성타운점 -> 9504 중복이므로 지우기
# 서울특별시 서초구 서초대로50길 19	    37.492403	127.013162 -> 교대역점

In [20]:
# 서울특별시 서초구 서초대로50길 19	    37.492403	127.013162 -> 교대역점      -> 위도: 37.4923947, 경도: 127.0130463
# 서울특별시 강서구 마곡동 797-9	    37.559854	126.834848 -> 발산역점      -> 위도: 37.5597121, 경도: 126.8348083
# 서울특별시 양천구 신정중앙로 95	    37.526607	126.862547 -> 목동로데오점   -> 위도: 37.5265624, 경도: 126.8626309

### ***추가로 찾은 지점***
- 위도, 경도로 역지오코딩 해보니 정확도가 떨어짐 -> 매장 안나옴
- 그래서 웹 크롤링을 통해 위도, 경도 찾음

### **교대역점**
- 도로명: 서울 서초구 서초대로 50길 19
- 지번주소: 서울 서초구 서초동 1574-4
- 위도: 37.4923947, 경도: 127.0130463
### **발산역점**
- 도로명: 서울 강서구 마곡중앙6로 70
- 지번주소: 서울 강서구 마곡동 797-9
- 위도: 37.5597121, 경도: 126.8348083
### **목동로데오점**
- 도로명: 서울특별시 양천구 신정중앙로 95
- 지번주소: 서울특별시 양천구 신정동 887-8
- 위도: 37.5265624, 경도: 126.8626309

In [21]:
# 찾을 지번주소 목록 정의
target_jibun = [
    "서울특별시 서초구 서초동 1574-4",
    "서울특별시 강서구 마곡동 797-9",
    "서울특별시 양천구 신정동 887-8"
]

In [22]:
# 지번주소 기준 필터링
check_df = 금별맥주2[금별맥주2['지번주소'].isin(target_jibun)]

In [23]:
# 핵심 컬럼만 확인
check_df[['지점명', '지번주소', '도로명주소', '위도', '경도']]

지점명                  지번주소                 도로명주소         위도          경도
9488  NaN   서울특별시 양천구 신정동 887-8    서울특별시 양천구 신정중앙로 95  37.526607  126.862547
9648  NaN  서울특별시 서초구 서초동 1574-4  서울특별시 서초구 서초대로50길 19  37.492403  127.013162
9717  NaN   서울특별시 강서구 마곡동 797-9   서울특별시 강서구 마곡중앙6로 70  37.559854  126.834848

In [24]:
# 지번주소가 1건인지 확인
check_df['지번주소'].value_counts()

# 지번주소가 모두 1건씩인거 확인 완료
# 지번주소를 기준으로 3개지점 덮어씌우기

지번주소
서울특별시 양천구 신정동 887-8     1
서울특별시 서초구 서초동 1574-4    1
서울특별시 강서구 마곡동 797-9     1
Name: count, dtype: int64

In [25]:
update_map = {
    '서울특별시 서초구 서초동 1574-4': {
        '지점명': '교대역점',
        '도로명주소': '서울특별시 서초구 서초대로 50길 19',
        '위도': 37.4923947,
        '경도': 127.0130463
    },
    '서울특별시 강서구 마곡동 797-9': {
        '지점명': '발산역점',
        '도로명주소': '서울특별시 강서구 마곡중앙6로 70',
        '위도': 37.5597121,
        '경도': 126.8348083
    },
    '서울특별시 양천구 신정동 887-8': {
        '지점명': '목동로데오점',
        '도로명주소': '서울특별시 양천구 신정중앙로 95',
        '위도': 37.5265624,
        '경도': 126.8626309
    }
}

for jibun, values in update_map.items():
    mask = 금별맥주2['지번주소'] == jibun
    for col, val in values.items():
        금별맥주2.loc[mask, col] = val


In [26]:
# 업데이트 후 데이터 프레임 확인
check_jibun = [
    '서울특별시 서초구 서초동 1574-4',
    '서울특별시 강서구 마곡동 797-9',
    '서울특별시 양천구 신정동 887-8'
]

금별맥주2.loc[
    금별맥주2['지번주소'].isin(check_jibun),
    ['지점명', '도로명주소', '지번주소', '위도', '경도']
]


지점명                  도로명주소                  지번주소         위도  \
9488  목동로데오점     서울특별시 양천구 신정중앙로 95   서울특별시 양천구 신정동 887-8  37.526562   
9648    교대역점  서울특별시 서초구 서초대로 50길 19  서울특별시 서초구 서초동 1574-4  37.492395   
9717    발산역점    서울특별시 강서구 마곡중앙6로 70   서울특별시 강서구 마곡동 797-9  37.559712   

              경도  
9488  126.862631  
9648  127.013046  
9717  126.834808

In [27]:
금별맥주2.loc[
    금별맥주2['place_id'].duplicated(keep=False)
].sort_values('place_id')[['place_id', '상호명', '지점명','지번주소', '도로명주소','위도','경도']]

place_id          상호명     지점명                    지번주소  \
10178  1175386565         금별맥주     미금점    경기도 성남시 분당구 구미동 41-2   
9866   1175386565         금별맥주     NaN   경기도 성남시 분당구 정자동 159-4   
9634   1195530733         금별맥주    사당역점    서울특별시 서초구 방배동 444-16   
9759   1195530733         금별맥주     NaN       서울특별시 동작구 사당동 138   
9717   1249032365         금별맥주    발산역점     서울특별시 강서구 마곡동 797-9   
9740   1249032365     금별맥주마곡역점     NaN     서울특별시 강서구 마곡동 799-2   
9488   1265782586         금별맥주  목동로데오점     서울특별시 양천구 신정동 887-8   
9576   1265782586  금별맥주신정네거리역점     NaN      서울특별시 양천구 신정동 1326   
10334  1418180719         금별맥주     NaN       경기도 광명시 일직동 512-3   
10059  1418180719         금별맥주    광명역점       경기도 광명시 일직동 512-3   
10071  1473371787       금별맥주안산     중앙점  경기도 안산시 단원구 고잔동 539-18   
9883   1473371787         금별맥주     NaN   경기도 안산시 상록구 본오동 879-6   
10119  1473371787    금별맥주한대앞역점     NaN    경기도 안산시 상록구 이동 715-2   
9504   1494938619         금별맥주     NaN   서울특별시 서초구 서초동 1327-35   
9648   1494938619         금별맥주    교대역점    서울특별시 서초구 서초동 1574-4   

                         도로명주소         위도          경도  
10178  경기도 성남시 분당구 미금일로86번길 20  37.347801  127.110480  
9866      경기도 성남시 분당구 정자일로 162  37.364627  127.106305  
9634         서울특별시 서초구 동작대로 18  37.478103  126.982092  
9759     서울특별시 동작구 동작대로27나길 21  37.486608  126.980676  
9717       서울특별시 강서구 마곡중앙6로 70  37.559712  126.834808  
9740        서울특별시 강서구 공항대로 190  37.559034  126.828636  
9488        서울특별시 양천구 신정중앙로 95  37.526562  126.862631  
9576       서울특별시 양천구 중앙로43길 14  37.520421  126.851337  
10334           경기도 광명시 양지로 21  37.417687  126.882211  
10059           경기도 광명시 양지로 19  37.419012  126.882615  
10071      경기도 안산시 단원구 고잔1길 45  37.318396  126.838525  
9883       경기도 안산시 상록구 샘골로 165  37.298631  126.862918  
10119     경기도 안산시 상록구 광덕1로 379  37.308655  126.851248  
9504      서울특별시 서초구 서초대로78길 38  37.494526  127.028225  
9648     서울특별시 서초구 서초대로 50길 19  37.492395  127.013046

In [28]:
# 지점명이 비어있는 행만 대상으로
mask = 금별맥주2['지점명'].isna()

# 상호명에서 '금별맥주' 뒤 문자열 추출 → 지점명으로
금별맥주2.loc[mask, '지점명'] = (
    금별맥주2.loc[mask, '상호명']
    .str.replace('금별맥주', '', regex=False)
    .str.strip()
)

# 빈 문자열은 다시 NaN 처리 (안전장치)
금별맥주2.loc[
    금별맥주2['지점명'] == '',
    '지점명'
] = np.nan

In [29]:
금별맥주2.loc[
    금별맥주2['place_id'].duplicated(keep=False)
].sort_values('place_id')[['place_id', '상호명', '지점명','지번주소', '도로명주소','위도','경도']]

place_id          상호명      지점명                    지번주소  \
10178  1175386565         금별맥주      미금점    경기도 성남시 분당구 구미동 41-2   
9866   1175386565         금별맥주      NaN   경기도 성남시 분당구 정자동 159-4   
9634   1195530733         금별맥주     사당역점    서울특별시 서초구 방배동 444-16   
9759   1195530733         금별맥주      NaN       서울특별시 동작구 사당동 138   
9717   1249032365         금별맥주     발산역점     서울특별시 강서구 마곡동 797-9   
9740   1249032365     금별맥주마곡역점     마곡역점     서울특별시 강서구 마곡동 799-2   
9488   1265782586         금별맥주   목동로데오점     서울특별시 양천구 신정동 887-8   
9576   1265782586  금별맥주신정네거리역점  신정네거리역점      서울특별시 양천구 신정동 1326   
10334  1418180719         금별맥주      NaN       경기도 광명시 일직동 512-3   
10059  1418180719         금별맥주     광명역점       경기도 광명시 일직동 512-3   
10071  1473371787       금별맥주안산      중앙점  경기도 안산시 단원구 고잔동 539-18   
9883   1473371787         금별맥주      NaN   경기도 안산시 상록구 본오동 879-6   
10119  1473371787    금별맥주한대앞역점    한대앞역점    경기도 안산시 상록구 이동 715-2   
9504   1494938619         금별맥주      NaN   서울특별시 서초구 서초동 1327-35   
9648   1494938619         금별맥주     교대역점    서울특별시 서초구 서초동 1574-4   

                         도로명주소         위도          경도  
10178  경기도 성남시 분당구 미금일로86번길 20  37.347801  127.110480  
9866      경기도 성남시 분당구 정자일로 162  37.364627  127.106305  
9634         서울특별시 서초구 동작대로 18  37.478103  126.982092  
9759     서울특별시 동작구 동작대로27나길 21  37.486608  126.980676  
9717       서울특별시 강서구 마곡중앙6로 70  37.559712  126.834808  
9740        서울특별시 강서구 공항대로 190  37.559034  126.828636  
9488        서울특별시 양천구 신정중앙로 95  37.526562  126.862631  
9576       서울특별시 양천구 중앙로43길 14  37.520421  126.851337  
10334           경기도 광명시 양지로 21  37.417687  126.882211  
10059           경기도 광명시 양지로 19  37.419012  126.882615  
10071      경기도 안산시 단원구 고잔1길 45  37.318396  126.838525  
9883       경기도 안산시 상록구 샘골로 165  37.298631  126.862918  
10119     경기도 안산시 상록구 광덕1로 379  37.308655  126.851248  
9504      서울특별시 서초구 서초대로78길 38  37.494526  127.028225  
9648     서울특별시 서초구 서초대로 50길 19  37.492395  127.013046

In [30]:
금별맥주3 = 금별맥주2.loc[
    금별맥주2['place_id'].duplicated(keep=False)
].copy()

In [31]:
# 행/열 구조 확인
print(금별맥주3.shape)
print(금별맥주3.columns)

금별맥주3.loc[
    금별맥주3['place_id'].duplicated(keep=False)
].sort_values('place_id')[['place_id', '상호명', '지점명','지번주소', '도로명주소','위도','경도']]


(15, 34)
Index(['상가업소번호', 'brand', '상호명', '지점명', '상권업종대분류코드', '상권업종대분류명', '상권업종중분류코드',
       '상권업종중분류명', '상권업종소분류코드', '상권업종소분류명', '표준산업분류코드', '표준산업분류명', '시도코드',
       '시도명', '시군구코드', '시군구명', '행정동코드', '행정동명', '법정동코드', '법정동명', '지번코드',
       '지번본번지', '지번부번지', '지번주소', '도로명코드', '도로명', '도로명주소', '층정보', '경도', '위도',
       'place_id', '월간 검색량', '블로그누적발행', 'encoded'],
      dtype='object')


place_id          상호명      지점명                    지번주소  \
10178  1175386565         금별맥주      미금점    경기도 성남시 분당구 구미동 41-2   
9866   1175386565         금별맥주      NaN   경기도 성남시 분당구 정자동 159-4   
9634   1195530733         금별맥주     사당역점    서울특별시 서초구 방배동 444-16   
9759   1195530733         금별맥주      NaN       서울특별시 동작구 사당동 138   
9717   1249032365         금별맥주     발산역점     서울특별시 강서구 마곡동 797-9   
9740   1249032365     금별맥주마곡역점     마곡역점     서울특별시 강서구 마곡동 799-2   
9488   1265782586         금별맥주   목동로데오점     서울특별시 양천구 신정동 887-8   
9576   1265782586  금별맥주신정네거리역점  신정네거리역점      서울특별시 양천구 신정동 1326   
10334  1418180719         금별맥주      NaN       경기도 광명시 일직동 512-3   
10059  1418180719         금별맥주     광명역점       경기도 광명시 일직동 512-3   
10071  1473371787       금별맥주안산      중앙점  경기도 안산시 단원구 고잔동 539-18   
9883   1473371787         금별맥주      NaN   경기도 안산시 상록구 본오동 879-6   
10119  1473371787    금별맥주한대앞역점    한대앞역점    경기도 안산시 상록구 이동 715-2   
9504   1494938619         금별맥주      NaN   서울특별시 서초구 서초동 1327-35   
9648   1494938619         금별맥주     교대역점    서울특별시 서초구 서초동 1574-4   

                         도로명주소         위도          경도  
10178  경기도 성남시 분당구 미금일로86번길 20  37.347801  127.110480  
9866      경기도 성남시 분당구 정자일로 162  37.364627  127.106305  
9634         서울특별시 서초구 동작대로 18  37.478103  126.982092  
9759     서울특별시 동작구 동작대로27나길 21  37.486608  126.980676  
9717       서울특별시 강서구 마곡중앙6로 70  37.559712  126.834808  
9740        서울특별시 강서구 공항대로 190  37.559034  126.828636  
9488        서울특별시 양천구 신정중앙로 95  37.526562  126.862631  
9576       서울특별시 양천구 중앙로43길 14  37.520421  126.851337  
10334           경기도 광명시 양지로 21  37.417687  126.882211  
10059           경기도 광명시 양지로 19  37.419012  126.882615  
10071      경기도 안산시 단원구 고잔1길 45  37.318396  126.838525  
9883       경기도 안산시 상록구 샘골로 165  37.298631  126.862918  
10119     경기도 안산시 상록구 광덕1로 379  37.308655  126.851248  
9504      서울특별시 서초구 서초대로78길 38  37.494526  127.028225  
9648     서울특별시 서초구 서초대로 50길 19  37.492395  127.013046

### ***현재상태***
- 금별맥주2 : 원본 복사 + 일부 중복 place_id 정제 전
- 금별맥주3 : 금별맥주2 중 중복 place_id만 추출해서 작업 중(삭제 대상 포함)
- 즉, 금별맥주3은 금별맥주2의 부분집합이고, 삭제할 행도 금별맥주3 안에만 있음

### ***삭제 후 금별맥주4 만들기***
- 지금 이 단계는 금별맥주3에 있던 중복 place_id 데이터만 남은 상태
- 즉, 금별맥주2의 다른 행들은 포함되어 있지 않음

- place_id 중복 2개 발견
    - 1249032365 : 마곡역점, 발산역점
    - 1265782586 : 신정네거리역점, 목동로데오점

- 웹 크롤링 통해 place_id 중복 없애기

### ***금별맥주2 내용 합치기***
- 금별맥주2 + 금별맥주4 = 전체 데이터
- 금별맥주2에서 중복 place_id 제거(금별맥주3에 포함된 행 제외) 후 금별맥주 4와 나머지 합치기
- 이렇게 하면 중복 place_id만 작업한 금별맥주3 + 나머지 금별맥주2가 합쳐져서 전체 데이터셋이 됨
- 중복 처리 및 NaN 삭제가 반영된 완전한 금별맥주4 완성

### ***검증***
- 중복 place_id가 잘 처리됐는지 확인

In [32]:
# 금별맥주3(place_id가 중복)에서 지점명이 NaN이거나 지점명 "중앙점"인 행 삭제
삭제대상 = 금별맥주3.loc[
    금별맥주3['지점명'].isna() | (금별맥주3['지점명'] == "중앙점")
]


In [33]:
# 삭제 후 금별맥주 4에 저장
# 원본 금별맥주3은 그대로 유지되고, 삭제된 복사본이 금별맥주4가 됨
금별맥주4 = 금별맥주3.drop(삭제대상.index)

삭제대상[['place_id', '상호명', '지점명']]

place_id     상호명  지점명
9504   1494938619    금별맥주  NaN
9759   1195530733    금별맥주  NaN
9866   1175386565    금별맥주  NaN
9883   1473371787    금별맥주  NaN
10071  1473371787  금별맥주안산  중앙점
10334  1418180719    금별맥주  NaN

In [34]:
# 지점명이 NaN이거나 '중앙점'인 행이 남아있는지 확인
금별맥주4.loc[금별맥주4['지점명'].isna() | (금별맥주4['지점명'] == "중앙점")]

Empty DataFrame
Columns: [상가업소번호, brand, 상호명, 지점명, 상권업종대분류코드, 상권업종대분류명, 상권업종중분류코드, 상권업종중분류명, 상권업종소분류코드, 상권업종소분류명, 표준산업분류코드, 표준산업분류명, 시도코드, 시도명, 시군구코드, 시군구명, 행정동코드, 행정동명, 법정동코드, 법정동명, 지번코드, 지번본번지, 지번부번지, 지번주소, 도로명코드, 도로명, 도로명주소, 층정보, 경도, 위도, place_id, 월간 검색량, 블로그누적발행, encoded]
Index: []

In [35]:
# 중복 place_id가 있는지 확인
금별맥주4['place_id'].duplicated().sum()

np.int64(2)

In [36]:
# 중복 place_id 확인
금별맥주4.loc[
    금별맥주4['place_id'].duplicated(keep=False)  # 중복 place_id 모두 선택
].sort_values('place_id')[  # place_id 기준으로 정렬
    ['place_id', '상호명', '지점명','지번주소', '도로명주소','위도','경도']  # 확인할 컬럼 선택
]


place_id          상호명      지점명                 지번주소  \
9740  1249032365     금별맥주마곡역점     마곡역점  서울특별시 강서구 마곡동 799-2   
9717  1249032365         금별맥주     발산역점  서울특별시 강서구 마곡동 797-9   
9576  1265782586  금별맥주신정네거리역점  신정네거리역점   서울특별시 양천구 신정동 1326   
9488  1265782586         금별맥주   목동로데오점  서울특별시 양천구 신정동 887-8   

                    도로명주소         위도          경도  
9740   서울특별시 강서구 공항대로 190  37.559034  126.828636  
9717  서울특별시 강서구 마곡중앙6로 70  37.559712  126.834808  
9576  서울특별시 양천구 중앙로43길 14  37.520421  126.851337  
9488   서울특별시 양천구 신정중앙로 95  37.526562  126.862631

In [37]:
# place_id 중복 2개 발견
# 1249032365 : 마곡역점, 발산역점
# 1265782586 : 신정네거리역점, 목동로데오점

# 웹 크롤링 통해 place_id 중복 없애기 (bizId로 찾기)
# 1249032365 : 마곡역점
# 1265782586 : 신정네거리역점
# 1215140348 : 발산역점
# 1875835175 : 목동로데오점

In [38]:
# 발산역점 place_id 수정
금별맥주4.loc[금별맥주4['지점명'] == "발산역점", 'place_id'] = 1215140348

# 목동로데오점 place_id 수정
금별맥주4.loc[금별맥주4['지점명'] == "목동로데오점", 'place_id'] = 1875835175


In [39]:
# 중복 place_id 재확인
금별맥주4.loc[
    금별맥주4['place_id'].duplicated(keep=False)  # 중복 place_id 모두 선택
].sort_values('place_id')[  # place_id 기준으로 정렬
    ['place_id', '상호명', '지점명','지번주소', '도로명주소','위도','경도']  # 확인할 컬럼 선택
]

# 중복 place_id 없는거 확인 완료

Empty DataFrame
Columns: [place_id, 상호명, 지점명, 지번주소, 도로명주소, 위도, 경도]
Index: []

In [40]:
# 중복 place_id 삭제 한 데이터 (지점명 NaN, 중앙점은 금별맥주3에서 삭제함)
금별맥주4[['place_id', '상호명', '지점명', '지번주소', '도로명주소', '위도', '경도']]

place_id          상호명      지점명                  지번주소  \
9488   1875835175         금별맥주   목동로데오점   서울특별시 양천구 신정동 887-8   
9576   1265782586  금별맥주신정네거리역점  신정네거리역점    서울특별시 양천구 신정동 1326   
9634   1195530733         금별맥주     사당역점  서울특별시 서초구 방배동 444-16   
9648   1494938619         금별맥주     교대역점  서울특별시 서초구 서초동 1574-4   
9717   1215140348         금별맥주     발산역점   서울특별시 강서구 마곡동 797-9   
9740   1249032365     금별맥주마곡역점     마곡역점   서울특별시 강서구 마곡동 799-2   
10059  1418180719         금별맥주     광명역점     경기도 광명시 일직동 512-3   
10119  1473371787    금별맥주한대앞역점    한대앞역점  경기도 안산시 상록구 이동 715-2   
10178  1175386565         금별맥주      미금점  경기도 성남시 분당구 구미동 41-2   

                         도로명주소         위도          경도  
9488        서울특별시 양천구 신정중앙로 95  37.526562  126.862631  
9576       서울특별시 양천구 중앙로43길 14  37.520421  126.851337  
9634         서울특별시 서초구 동작대로 18  37.478103  126.982092  
9648     서울특별시 서초구 서초대로 50길 19  37.492395  127.013046  
9717       서울특별시 강서구 마곡중앙6로 70  37.559712  126.834808  
9740        서울특별시 강서구 공항대로 190  37.559034  126.828636  
10059           경기도 광명시 양지로 19  37.419012  126.882615  
10119     경기도 안산시 상록구 광덕1로 379  37.308655  126.851248  
10178  경기도 성남시 분당구 미금일로86번길 20  37.347801  127.110480

### ***현재 목표***
- 금별맥주2 + 금별맥주4 = 정제완료 한 금별맥주5 만들기
- 중복 place_id 없어야 함
- 일부 지점명이 NaN인건 괜찮음

- 금별맥주2
    - 원본 데이터 + 일부 중복 place_id 포함
    - 중복 place_id를 추출해서 금별맥주3에서 정제 작업 시작

- 금별맥주4
    - 금별맥주3에서 중복 place_id 중 불필요한 행 삭제 + place_id 수정 완료
    - 즉, 중복 place_id 정제 완료본

**- 주의할 점**
- 금별맥주4는 금별맥주2에서 추출한 행들을 정제한 결과물
- 따라서 금별맥주2에서 금별맥주4에서 처리한 동일한 행(같은 place_id) 을 그대로 두면 중복 발생
- 그래서 금별맥주2에서 금별맥주4에 포함된 place_id 행을 제거하고 합쳐야 함 → 중복 방지

In [41]:
# 금별맥주4에 있는 place_id를 금별맥주2에서 제거
금별맥주2_remaining = 금별맥주2[~금별맥주2['place_id'].isin(금별맥주4['place_id'])]

In [42]:
# 남은 금별맥주2 + 정제된 금별맥주4 합치기
금별맥주5 = pd.concat([금별맥주2_remaining, 금별맥주4], ignore_index=True)

In [43]:
# 중복 검증
print("place_id 중복 검증(0이면 중복 제거 완료):", 금별맥주5['place_id'].duplicated().sum())  # 0이면 중복 제거 완료

place_id 중복 검증(0이면 중복 제거 완료): 0


In [44]:
행개수 = 금별맥주5.shape[0]
print("금별맥주5(정제완료) 행 개수:",행개수)

금별맥주5[['place_id', '상호명', '지점명', '지번주소', '도로명주소', '위도', '경도']]

금별맥주5(정제완료) 행 개수: 112


place_id            상호명        지점명                      지번주소  \
0    1113384749     금별맥주동양미래대점     동양미래대점      서울특별시 구로구 고척동 62-134   
1    1320051621      금별맥주양천향교점      양천향교점         서울특별시 강서구 마곡동 776   
2    1588254549           금별맥주        NaN       서울특별시 송파구 문정동 643-1   
3    1601981946       금별맥주송정역점       송정역점      서울특별시 강서구 공항동 53-109   
4    1161335521           금별맥주        NaN     서울특별시 서대문구 창천동 52-158   
5    1126343564           금별맥주       연신내점      서울특별시 은평구 갈현동 399-13   
6    1632711633           금별맥주      성신여대점       서울특별시 성북구 동선동1가 4-4   
7    1516484705         금별맥주신길       삼거리점       서울특별시 영등포구 신길동 99-7   
8       9650107         금별맥주서래        마을점       서울특별시 서초구 반포동 92-14   
9    1308648776           금별맥주        성수점   서울특별시 성동구 성수동2가 289-273   
10   1972585179           금별맥주        NaN     서울특별시 관악구 신림동 1640-13   
11   1189283598           금별맥주       신도림점          서울특별시 구로구 구로동 30   
12   1120915524      금별맥주종로평창점      종로평창점       서울특별시 종로구 평창동 289-5   
13   1831537710           금별맥주        홍제점      서울특별시 서대문구 홍제동 270-1   
14   1456505357           금별맥주        홍대점      서울특별시 마포구 서교동 395-18   
15      9230108       금별맥주경희대점       경희대점      서울특별시 동대문구 회기동 60-55   
16   1765329714           금별맥주        길동점        서울특별시 강동구 길동 413-1   
17   1005671374         올라금별맥주        당산점    서울특별시 영등포구 당산동6가 332-3   
18   1109103798         금별맥주응암       오거리점      서울특별시 은평구 응암동 591-12   
19   1654545522           금별맥주       영등포점   서울특별시 영등포구 영등포동3가 11-18   
20      9500103      금별맥주까치산역점      까치산역점       서울특별시 강서구 화곡동 346-3   
21   1151197342      금별맥주충정로역점      충정로역점          서울특별시 중구 중림동 200   
22   1480935223       금별맥주등촌역점       등촌역점       서울특별시 양천구 목동 609-24   
23   1294998541         금별맥주강서        구청점     서울특별시 강서구 화곡동 1119-10   
24   1164697805           금별맥주       오목교점      서울특별시 양천구 목동 406-184   
25   1460937600         금별맥주묵동        자이점        서울특별시 중랑구 묵동 164-4   
26   1826878021           금별맥주        NaN       서울특별시 송파구 방이동 59-10   
27   1035417750           금별맥주   구로디지털단지점     서울특별시 구로구 구로동 1124-38   
28   1778168403           금별맥주        화곡점      서울특별시 강서구 화곡동 98-235   
29   1269875819           금별맥주        응암점       서울특별시 은평구 응암동 87-18   
30   1058646920           금별맥주        천호점       서울특별시 강동구 천호동 412-2   
31   1754035019           금별맥주        종로점        서울특별시 종로구 관철동 19-4   
32   1538729455          금별맥주점      마곡나루역       서울특별시 강서구 마곡동 760-2   
33   1160722316           금별맥주        건대점        서울특별시 광진구 화양동 9-88   
34   1117014686         금별맥주마포        구청점       서울특별시 마포구 성산동 592-2   
35      9215103           금별맥주        구의점      서울특별시 광진구 구의동 257-93   
36   1390537313           금별맥주        NaN       서울특별시 강남구 신사동 517-6   
37   1516621298         금별맥주영통        구청점    경기도 수원시 영통구 매탄동 1263-5   
38   1479645620         금별맥주시흥        은계점           경기도 시흥시 은행동 644   
39   1384001087     금별맥주평내호평역점     평내호평역점        경기도 남양주시 호평동 654-3   
40   1866157196         금별맥주수원        탑동점      경기도 수원시 권선구 탑동 861-1   
41   1821667598           금별맥주        서현점     경기도 성남시 분당구 서현동 247-2   
42   1326207551      금별맥주용인역북점      용인역북점     경기도 용인시 처인구 역북동 754-1   
43   1119031588           금별맥주        NaN    경기도 수원시 영통구 영통동 1010-2   
44   1548256786           금별맥주        백석점  경기도 고양시 일산동구 백석동 1282-10   
45   1736872254           금별맥주        NaN    경기도 평택시 포승읍 도곡리 1145-8   
46   1504742739           금별맥주      용인보정점      경기도 용인시 기흥구 보정동 1262   
47   1021328867           금별맥주        야당점          경기도 파주시 야당동 1085   
48      2360112        금별맥주다산점        다산점         경기도 남양주시 다산동 6061   
49   1005285987           금별맥주       산본역점          경기도 군포시 산본동 1129   
50   2035205067           금별맥주        NaN         경기도 시흥시 장현동 532-1   
51   1878005111           금별맥주       수원대점      경기도 화성시 봉담읍 와우리 43-3   
52   1206922924      금별맥주오산궐동점      오산궐동점          경기도 오산시 궐동 680-2   
53   1473086261           금별맥주        구래점        경기도 김포시 구래동 6882-8   
54   1649261391           금별맥주        화정점     경기도 고양시 덕양구 화정동 983-1   
55   1636872723           금별맥주 

In [45]:
금별맥주5

상가업소번호 brand            상호명        지점명 상권업종대분류코드 상권업종대분류명  \
0    MA010120220804156426  금별맥주     금별맥주동양미래대점     동양미래대점        I2       음식   
1    MA010120220808688996  금별맥주      금별맥주양천향교점      양천향교점        I2       음식   
2    MA010120220806941672  금별맥주           금별맥주        NaN        I2       음식   
3    MA010120220810164671  금별맥주       금별맥주송정역점       송정역점        I2       음식   
4    MA010120220812201759  금별맥주           금별맥주        NaN        I2       음식   
5    MA0101202501A0025382  금별맥주           금별맥주       연신내점        I2       음식   
6    MA0101202503A0041391  금별맥주           금별맥주      성신여대점        I2       음식   
7    MA0101202412A0059403  금별맥주         금별맥주신길       삼거리점        I2       음식   
8    MA0101202411A0018154  금별맥주         금별맥주서래        마을점        I2       음식   
9    MA010120220807891737  금별맥주           금별맥주        성수점        I2       음식   
10   MA010120220805757261  금별맥주           금별맥주        NaN        I2       음식   
11   MA0101202305A0139131  금별맥주           금별맥주       신도림점        I2       음식   
12   MA0101202307A0096441  금별맥주      금별맥주종로평창점      종로평창점        I2       음식   
13   MA0101202310A0090370  금별맥주           금별맥주        홍제점        I2       음식   
14   MA0101202311A0031574  금별맥주           금별맥주        홍대점        I2       음식   
15   MA0101202311A0006903  금별맥주       금별맥주경희대점       경희대점        I2       음식   
16   MA010120220814163743  금별맥주           금별맥주        길동점        I2       음식   
17   MA0101202212A0039390  금별맥주         올라금별맥주        당산점        I2       음식   
18   MA0101202210A0009834  금별맥주         금별맥주응암       오거리점        I2       음식   
19   MA0101202210A0027366  금별맥주           금별맥주       영등포점        I2       음식   
20   MA0101202209A0041617  금별맥주      금별맥주까치산역점      까치산역점        I2       음식   
21   MA0101202210A0003451  금별맥주      금별맥주충정로역점      충정로역점        I2       음식   
22   MA0101202302A0105619  금별맥주       금별맥주등촌역점       등촌역점        I2       음식   
23   MA0101202303A0071486  금별맥주         금별맥주강서        구청점        I2       음식   
24   MA0101202305A0064259  금별맥주           금별맥주       오목교점        I2       음식   
25   MA0101202305A0172560  금별맥주         금별맥주묵동        자이점        I2       음식   
26   MA0101202305A0049354  금별맥주           금별맥주        NaN        I2       음식   
27   MA0101202303A0116482  금별맥주           금별맥주   구로디지털단지점        I2       음식   
28   MA0101202305A0038991  금별맥주           금별맥주        화곡점        I2       음식   
29   MA0101202302A0018641  금별맥주           금별맥주        응암점        I2       음식   
30   MA0101202302A0081428  금별맥주           금별맥주        천호점        I2       음식   
31   MA0101202210A0020762  금별맥주           금별맥주        종로점        I2       음식   
32   MA0101202211A0097249  금별맥주          금별맥주점      마곡나루역        I2       음식   
33   MA0101202210A0045520  금별맥주           금별맥주        건대점        I2       음식   
34   MA010120220809880201  금별맥주         금별맥주마포        구청점        I2       음식   
35   MA0101202209A0035602  금별맥주           금별맥주        구의점        I2       음식   
36   MA010120220810424872  금별맥주           금별맥주        NaN        I2       음식   
37   MA0101202209A0062760  금별맥주         금별맥주영통        구청점        I2       음식   
38   MA0101202209A0063723  금별맥주         금별맥주시흥        은계점        I2       음식   
39   MA010120220807074853  금별맥주     금별맥주평내호평역점     평내호평역점        I2       음식   
40   MA010120220812130051  금별맥주         금별맥주수원        탑동점        I2       음식   
41   MA010120220811500121  금별맥주           금별맥주        서현점        I2       음식   
42   MA010120220808091315  금별맥주      금별맥주용인역북점      용인역북점        I2       음식   
43   MA010120220806894330  금별맥주           금별맥주        NaN        I2       음식   
44   MA010120220805090548  금별맥주           금별맥주        백석점        I2       음식   
45   MA010120220808636259  금별맥주           금별맥주        NaN        I2       음식   
46   MA010120220806437848  금별맥주           금별맥주      용인보정점        I2       음식   
47   MA010120220811967626  금별맥주           금별맥주        야당점        I2       음식   
48   MA010120220805986182  금별맥주        금별맥주다산점        다산점        I2       음식   
49   MA010120220813

In [47]:
금별맥주5.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112 entries, 0 to 111
Data columns (total 34 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   상가업소번호     112 non-null    object 
 1   brand      112 non-null    object 
 2   상호명        112 non-null    object 
 3   지점명        94 non-null     object 
 4   상권업종대분류코드  112 non-null    object 
 5   상권업종대분류명   112 non-null    object 
 6   상권업종중분류코드  112 non-null    object 
 7   상권업종중분류명   112 non-null    object 
 8   상권업종소분류코드  112 non-null    object 
 9   상권업종소분류명   112 non-null    object 
 10  표준산업분류코드   112 non-null    object 
 11  표준산업분류명    112 non-null    object 
 12  시도코드       112 non-null    int64  
 13  시도명        112 non-null    object 
 14  시군구코드      112 non-null    int64  
 15  시군구명       112 non-null    object 
 16  행정동코드      112 non-null    int64  
 17  행정동명       112 non-null    object 
 18  법정동코드      112 non-null    int64  
 19  법정동명       112 non-null    object 
 20  지번코드      

### ***현재 상황***
- 데이터 갯수 : 118 -> 112 (6개 감소)
- 중복된 place_id, 폐점 매장 제거완료
- 정제 과정에서 폐점 매장은 지점명 기준으로 삭제 했기에 상호명에 지점명이 포함되어 있는 경우 분리해서 지점명 채움
    - 지점명 62 -> 94 (32개 증가)
- place_id dtype : int 확인 완료

In [48]:
금별맥주5.to_csv(
    '금별맥주_정제완료.csv',
    index=False,
    encoding='utf-8-sig'
)